source/slang/slang-modifier-defs.h

```diff
@@ -28,16 +28,26 @@ SIMPLE_MODIFIER(GloballyCoherent)
 #undef SIMPLE_MODIFIER
 
 // A modifier that marks something as an operation that
+// translates to special
 // has a one-to-one translation to the IR, and thus
 // has no direct definition in the high-level language.
 //
 SYNTAX_CLASS(IntrinsicOpModifier, Modifier)
 
-    // token that names the intrinsic op
+    // Token that names the intrinsic op.
     FIELD(Token, opToken)
 
-    // The opcode for the intrinsic operation
-    FIELD_INIT(IROp, op, kIROp_Nop)
+    // The opcode for the intrinsic operation.
+    //
+    // If greather than or equal to zero, then `op`
+    // is an `IROp` and directly identifies an IR
+    // instruction that the intrinsic should translate to.
+    //
+    // If less than zero, then `op` is a `CompoundIntrinsicOp`
+    // which maps to zero or more IR instructions using
+    // special-case logic in the IR lowering phase.
+    //
+    FIELD_INIT(int32_t, op, 0)
```

Comment:
Perhaps would be preferable to have a typedef naming the Op than just the in32_t. Could also use an empty enum class, and a couple of free functions to determine what it is, and extract from (with asserts on invalid extractions).

Reply:
I might do the `typedef` - that was something I thought about when writing the code but didn't end up doing. Actually adding strong semantics and so forth seems like overkill given just how few points in the code need to worry about this.